plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/api/ApiTaskInterface.java

```diff
@@ -593,6 +629,35 @@ public void putTaskDoc(final KiaApiTaskDoc doc) throws Throwable {
         }
     }
 
+    public void claimTask(KieApiClaimTask claimTask) throws ApiException {
+
+        final String configId = claimTask.getConfigId();
+        if (null != configId) {
+            try {
+                final BpmWidgetInfo bpmWidgetInfo = bpmWidgetInfoManager.getBpmWidgetInfo(Integer.parseInt(configId));
+                final String information = bpmWidgetInfo.getInformationDraft();
+                if (StringUtils.isNotEmpty(information)) {
+                    final ApsProperties config = new ApsProperties();
+                    config.loadFromXml(information);
+                    String knowledgetSource = (String) config.get(KieBpmSystemConstants.WIDGET_INFO_PROP_KIE_SOURCE_ID);
+                    KieBpmConfig bpmConfig = this.getKieFormManager().getKieServerConfigurations().get(knowledgetSource);
+                    String containerId = config.getProperty("containerId");
+                    String taskId = claimTask.getTaskId();
+                    
+                    //TODO Check the user passed to the API
+                    String username = bpmConfig.getUsername();
```

Comment:
@alepintus We need to update this to use the logged in user if there is one. Same change for 5.0.3

Reply:
@joewhite101 Can I create a new task on Jira for this change?

Reply:
@alepintus Yeah. We can merge this without that change as long as we get a Jira opened.

Reply:
@joewhite101 new task created